### ISSUE TITLE
Tracking: stuff missing on imported EKS clusters

### ISSUE THREAD

**Issue by piontec** [giantswarm/roadmap#2849]

**Comment by piontec** [comment]
- [ ] add `giantswarm/critical` and potentially other `PodPriorities`

**Comment by weatherhog** [comment]
@mproffitt

**Comment by mcharriere** [comment]
For external-dns we need access to route53 and be able to edit the WC DNS zone. On our clusters this is usually done with the `<cluster_id>-Route53Manager-Role`; which is created during cluster creation. 

In the case of imported EKS, this is a bit tricky because the use-cases can vary from customer to customer. 

For the `sample-customer` WC, we already have a DNS zone created but the role is missing. We could manually create the role; but I'm wondering if there is anything we can do to make it more reliable. 

Would it be possible to use crossplane for this? We'd need the CRDs from the `iam.aws.crossplane.io/v1beta1` group.

ref #2671

**Comment by mproffitt** [comment]
@mcharriere As this is a required cloud resource then yes, crossplane is probably the correct thing to apply it, however my opinion here is that this would potentially be custom as not all customers would want, or require our DNS.

IMO I think we need a separate issue to track how we handle the creation of custom resources as conditonal logic in crossplane requires custom composition functions to be built. cc: @weatherhog @puja108

**Comment by puja108** [comment]
Honestly, for now I would assume that us bringing our own DNS here, so at least there's something working ootb without having to rely on customer configuration would be a net benefit. The customer can still add their additional DNS zones, and the good thing with having our DNS with our ingress controller would be that they could start working with it ootb and with the same documentation that we already have. It also decoupled our functionality from any customer approval processes (DNS and certs can often delay setups immensly).

That said I do not know how much work such an added feature would be for you Martin, but it would make testing our own things on these externally-managed clusters (without customers) a lot easier.

Otherwise we'd need a dirty workaround to just unblock testing in Cabbage.

**Comment by mcharriere** [comment]
I managed to manually create the role I need, but it'd still need a `OIDC identity provider` configured on the cluster. This is probably a requirement for any app that needs access to the underlying infra.

**Comment by mproffitt** [comment]
There is already an oidc provider against this cluster. If you need the issuer URL you can get this with 

```bash
k get clusters.eks.aws.upbound.io sample-customer -o jsonpath='{.status.atProvider.identity[0].oidc[0].issuer}'
```

**Comment by mcharriere** [comment]
@mproffitt aha! that did the trick! thank you!

**Comment by QuentinBisson** [comment]
We're also missing CSI feature to test our apps that need CSI (a.k.a Loki) cc @QuantumEnigmaa

**Comment by mproffitt** [comment]
@QuentinBisson I've added the EBS block storage controller to `snail : sample-customer` EKS cluster

```nohighlight
k get csidrivers.storage.k8s.io
NAME              ATTACHREQUIRED   PODINFOONMOUNT   STORAGECAPACITY   TOKENREQUESTS   REQUIRESREPUBLISH   MODES        AGE
ebs.csi.aws.com   true             false            false             <unset>         false               Persistent   36m
efs.csi.aws.com   false            false            false             <unset>         false               Persistent   145d
```

**Comment by QuentinBisson** [comment]
Thank you :)

**Comment by weatherhog** [comment]
@mproffitt @piontec can this be closed?

**Comment by piontec** [comment]
"yes" for me

**Comment by mproffitt** [comment]
Closing as complete
